docs/guides/database.rst

```diff
@@ -193,9 +215,21 @@ $type
 $subtype
 	Entity subtype, e.g. ``'blog'`` or ``'page'``.
 
-By convention entities that have an uploaded avatar or icon will have the ``icontime`` property
-assigned. This means that you can use ``$entity->icontime`` to check if an icon exists for the given
-entity.
+Icons are assigned a timestamp when they're created. You can use ``$object->getIconTs($size)`` to 
+get the timestamp if you need to track changes to the icon.
+
+To check if an icon is set, use ``$object->hasIcon($size)``.
+
+Icon methods support passing an icon type if an entity has more than one icon. For example, a user
+might have an avatar and a cover photo icon. You would pass ``'cover_photo'`` as the icon type:
+
+.. code:: php
+	$object->setIconFromUploadedFile('uploaded_photo', 'cover_photo');
+
+	$object->getIconUrl([
+		'size' => 'medium',
+		'type' => 'cover_photo'
```

Comment:
When old sites get upgraded and modernized, clients often ask for "galleries of large images" (Pinterest is often mentioned as an example). The old image sizes are too small and clients are not even sure what sizes/shapes they want. So I end up having to add many new variations of them.

We should at least extend and redefine the list of default image sizes. My guess is that the current sizes were chosen because back then internet connections were remarkably slower. Image that was "'large" back then is quite small on today's standards.